classes/publication/Repository.php

```diff
@@ -165,6 +172,16 @@ public function version(Publication $publication): int
      */
     protected function setStatusOnPublish(Publication $publication)
     {
+        if ($publication->getData('continuousPublication')) {
+            $publication->setData('status', Submission::STATUS_PUBLISHED);
+            
+            if (!$publication->getData('datePublished')) {
+                $publication->setData('datePublished', Core::getCurrentDate());
+            }
+
+            return;
```

Comment:
Early return behaviour will make it likely that code added below this function will be inadvertently missed. It's better to avoid early returns in this kind of situation.